bundles/mapping/drawtools/plugin/DrawPlugin.ol3.js

```diff
@@ -208,22 +219,24 @@ Oskari.clazz.define(
             if(me._bufferedFeatureLayerId) {
                 bufferedFeatures = me.getFeatures(me._bufferedFeatureLayerId);
             }
-            if(me._shape === 'Circle') {
+            if(me._shape === 'Circle' && me._circleHasGeom) {
                 bufferedFeatures = me.getCircleAsPolygonFeature(features);
                 features = me.getCircleAsPointFeature(features);
             } else if(me._shape === 'LineString' && me._buffer > 0) {
                 me.addBufferPropertyToFeatures(features, me._buffer);
             }
             // TODO: get geojson for matching id
             var geojson = me.getFeaturesAsGeoJSON(features);
+            var crs = me._getSRS();
             var bufferedGeoJson = me.getFeaturesAsGeoJSON(bufferedFeatures);
 
             var data = {
                 length : me._length,
                 area : me._area,
                 buffer: me._buffer,
                 bufferedGeoJson: bufferedGeoJson,
-                shape: me._shape
+                shape: me._shape,
+                geometry: me.geometry

```

Comment:
Why is geometry added here and as parameter for event? It shouldn't be sent at all since it isn't serializable and geojson has the same info. Also CRS could be included in the geojson data.